# Analyzing Chinook music store database.

The main idea of this project is to showcase SQL skills answering business questions for imaginery Chinook music store using their [database](https://github.com/lerocha/chinook-database).

Here's a schema diagram for the Chinook database:

<img src='https://s3.amazonaws.com/dq-content/189/chinook-schema.svg' width="500" height="600">

In [1]:
# Import a library to use sql in Jupyter
import sqlalchemy

In [2]:
sqlalchemy.create_engine('sqlite:///chinook.db')
%load_ext sql
%sql sqlite:///chinook.db

'Connected: @chinook.db'

## Exploring the database

In [3]:
%%sql
SELECT COUNT(*)
  FROM sqlite_master
 WHERE type='table';

 * sqlite:///chinook.db
Done.


[(11,)]

There are 11 tables in our database. Let's explore each of them and also check the availability of views.

In [4]:
%%sql
SELECT *
  FROM sqlite_master
 WHERE type='table' OR type='view';

 * sqlite:///chinook.db
Done.


[('table', 'album', 'album', 2, 'CREATE TABLE [album]\n(\n    [album_id] INTEGER PRIMARY KEY NOT NULL,\n    [title] NVARCHAR(160)  NOT NULL,\n    [artist_id] INTEGER  NOT NULL,\n    FOREIGN KEY ([artist_id]) REFERENCES [artist] ([artist_id]) \n\t\tON DELETE NO ACTION ON UPDATE NO ACTION\n)'),
 ('table', 'artist', 'artist', 3, 'CREATE TABLE [artist]\n(\n    [artist_id] INTEGER PRIMARY KEY NOT NULL,\n    [name] NVARCHAR(120)\n)'),
 ('table', 'customer', 'customer', 4, 'CREATE TABLE [customer]\n(\n    [customer_id] INTEGER PRIMARY KEY NOT NULL,\n    [first_name] NVARCHAR(40)  NOT NULL,\n    [last_name] NVARCHAR(20)   ... (276 characters truncated) ...  [support_rep_id] INTEGER,\n    FOREIGN KEY ([support_rep_id]) REFERENCES [employee] ([employee_id]) \n\t\tON DELETE NO ACTION ON UPDATE NO ACTION\n)'),
 ('table', 'employee', 'employee', 5, 'CREATE TABLE [employee]\n(\n    [employee_id] INTEGER PRIMARY KEY NOT NULL,\n    [last_name] NVARCHAR(20)  NOT NULL,\n    [first_name] NVARCHAR(20)   ... (310 characters truncated) ... 4),\n    [email] NVARCHAR(60),\n    FOREIGN KEY ([reports_to]) REFERENCES [employee] ([employee_id]) \n\t\tON DELETE NO ACTION ON UPDATE NO ACTION\n)'),
 ('table', 'genre', 'genre', 6, 'CREATE TABLE [genre]\n(\n    [genre_id] INTEGER PRIMARY KEY NOT NULL,\n    [name] NVARCHAR(120)\n)'),
 ('table', 'invoice', 'invoice', 7, 'CREATE TABLE [invoice]\n(\n    [invoice_id] INTEGER PRIMARY KEY NOT NULL,\n    [customer_id] INTEGER  NOT NULL,\n    [invoice_date] DATETIME  NOT NUL ... (195 characters truncated) ... tal] NUMERIC(10,2)  NOT NULL,\n    FOREIGN KEY ([customer_id]) REFERENCES [customer] ([customer_id]) \n\t\tON DELETE NO ACTION ON UPDATE NO ACTION\n)'),
 ('table', 'invoice_line', 'invoice_line', 8, 'CREATE TABLE [invoice_line]\n(\n    [invoice_line_id] INTEGER PRIMARY KEY NOT NULL,\n    [invoice_id] INTEGER  NOT NULL,\n    [track_id] INTEGER  NOT ... (161 characters truncated) ...  DELETE NO ACTION ON UPDATE NO ACTION,\n    FOREIGN KEY ([track_id]) REFERENCES [track] ([track_id]) \n\t\tON DELETE NO ACTION ON UPDATE NO ACTION\n)'),
 ('table', 'media_type', 'media_type', 9, 'CREATE TABLE [media_type]\n(\n    [media_type_id] INTEGER PRIMARY KEY NOT NULL,\n    [name] NVARCHAR(120)\n)'),
 ('table', 'playlist', 'playlist', 10, 'CREATE TABLE [playlist]\n(\n    [playlist_id] INTEGER PRIMARY KEY NOT NULL,\n    [name] NVARCHAR(120)\n)'),
 ('table', 'playlist_track', 'playlist_track', 11, 'CREATE TABLE [playlist_track]\n(\n    [playlist_id] INTEGER  NOT NULL,\n    [track_id] INTEGER  NOT NULL,\n    CONSTRAINT [pk_playlist_track] PRIMARY ... (114 characters truncated) ...  DELETE NO ACTION ON UPDATE NO ACTION,\n    FOREIGN KEY ([track_id]) REFERENCES [track] ([track_id]) \n\t\tON DELETE NO ACTION ON UPDATE NO ACTION\n)'),
 ('table', 'track', 'track', 13, 'CREATE TABLE [track]\n(\n    [track_id] INTEGER PRIMARY KEY NOT NULL,\n    [name] NVARCHAR(200)  NOT NULL,\n    [album_id] INTEGER,\n    [media_type_ ... (377 characters truncated) ... ON ON UPDATE NO ACTION,\n    FOREIGN KEY ([media_type_id]) REFERENCES [media_type] ([media_type_id]) \n\t\tON DELETE NO ACTION ON UPDATE NO ACTION\n)')]

In [5]:
%%sql
SELECT *
  FROM employee
 LIMIT 10

 * sqlite:///chinook.db
Done.


[(1, 'Adams', 'Andrew', 'General Manager', None, '1962-02-18 00:00:00', '2016-08-14 00:00:00', '11120 Jasper Ave NW', 'Edmonton', 'AB', 'Canada', 'T5K 2N1', '+1 (780) 428-9482', '+1 (780) 428-3457', 'andrew@chinookcorp.com'),
 (2, 'Edwards', 'Nancy', 'Sales Manager', 1, '1958-12-08 00:00:00', '2016-05-01 00:00:00', '825 8 Ave SW', 'Calgary', 'AB', 'Canada', 'T2P 2T3', '+1 (403) 262-3443', '+1 (403) 262-3322', 'nancy@chinookcorp.com'),
 (3, 'Peacock', 'Jane', 'Sales Support Agent', 2, '1973-08-29 00:00:00', '2017-04-01 00:00:00', '1111 6 Ave SW', 'Calgary', 'AB', 'Canada', 'T2P 5M5', '+1 (403) 262-3443', '+1 (403) 262-6712', 'jane@chinookcorp.com'),
 (4, 'Park', 'Margaret', 'Sales Support Agent', 2, '1947-09-19 00:00:00', '2017-05-03 00:00:00', '683 10 Street SW', 'Calgary', 'AB', 'Canada', 'T2P 5G3', '+1 (403) 263-4423', '+1 (403) 263-4289', 'margaret@chinookcorp.com'),
 (5, 'Johnson', 'Steve', 'Sales Support Agent', 2, '1965-03-03 00:00:00', '2017-10-17 00:00:00', '7727B 41 Ave', 'Calgary', 'AB', 'Canada', 'T3B 1Y7', '1 (780) 836-9987', '1 (780) 836-9543', 'steve@chinookcorp.com'),
 (6, 'Mitchell', 'Michael', 'IT Manager', 1, '1973-07-01 00:00:00', '2016-10-17 00:00:00', '5827 Bowness Road NW', 'Calgary', 'AB', 'Canada', 'T3B 0C5', '+1 (403) 246-9887', '+1 (403) 246-9899', 'michael@chinookcorp.com'),
 (7, 'King', 'Robert', 'IT Staff', 6, '1970-05-29 00:00:00', '2017-01-02 00:00:00', '590 Columbia Boulevard West', 'Lethbridge', 'AB', 'Canada', 'T1K 5N8', '+1 (403) 456-9986', '+1 (403) 456-8485', 'robert@chinookcorp.com'),
 (8, 'Callahan', 'Laura', 'IT Staff', 6, '1968-01-09 00:00:00', '2017-03-04 00:00:00', '923 7 ST NW', 'Lethbridge', 'AB', 'Canada', 'T1H 1Y8', '+1 (403) 467-3351', '+1 (403) 467-8772', 'laura@chinookcorp.com')]

In [6]:
%%sql
SELECT *
  FROM customer
 LIMIT 10

 * sqlite:///chinook.db
Done.


[(1, 'Luís', 'Gonçalves', 'Embraer - Empresa Brasileira de Aeronáutica S.A.', 'Av. Brigadeiro Faria Lima, 2170', 'São José dos Campos', 'SP', 'Brazil', '12227-000', '+55 (12) 3923-5555', '+55 (12) 3923-5566', 'luisg@embraer.com.br', 3),
 (2, 'Leonie', 'Köhler', None, 'Theodor-Heuss-Straße 34', 'Stuttgart', None, 'Germany', '70174', '+49 0711 2842222', None, 'leonekohler@surfeu.de', 5),
 (3, 'François', 'Tremblay', None, '1498 rue Bélanger', 'Montréal', 'QC', 'Canada', 'H2G 1A7', '+1 (514) 721-4711', None, 'ftremblay@gmail.com', 3),
 (4, 'Bjørn', 'Hansen', None, 'Ullevålsveien 14', 'Oslo', None, 'Norway', '0171', '+47 22 44 22 22', None, 'bjorn.hansen@yahoo.no', 4),
 (5, 'František', 'Wichterlová', 'JetBrains s.r.o.', 'Klanova 9/506', 'Prague', None, 'Czech Republic', '14700', '+420 2 4172 5555', '+420 2 4172 5555', 'frantisekw@jetbrains.com', 4),
 (6, 'Helena', 'Holý', None, 'Rilská 3174/6', 'Prague', None, 'Czech Republic', '14300', '+420 2 4177 0449', None, 'hholy@gmail.com', 5),
 (7, 'Astrid', 'Gruber', None, 'Rotenturmstraße 4, 1010 Innere Stadt', 'Vienne', None, 'Austria', '1010', '+43 01 5134505', None, 'astrid.gruber@apple.at', 5),
 (8, 'Daan', 'Peeters', None, 'Grétrystraat 63', 'Brussels', None, 'Belgium', '1000', '+32 02 219 03 03', None, 'daan_peeters@apple.be', 4),
 (9, 'Kara', 'Nielsen', None, 'Sønder Boulevard 51', 'Copenhagen', None, 'Denmark', '1720', '+453 3331 9991', None, 'kara.nielsen@jubii.dk', 4),
 (10, 'Eduardo', 'Martins', 'Woodstock Discos', 'Rua Dr. Falcão Filho, 155', 'São Paulo', 'SP', 'Brazil', '01007-010', '+55 (11) 3033-5446', '+55 (11) 3033-4564', 'eduardo@woodstock.com.br', 4)]

In [7]:
%%sql
SELECT *
  FROM invoice
 LIMIT 10

 * sqlite:///chinook.db
Done.


[(1, 18, '2017-01-03 00:00:00', '627 Broadway', 'New York', 'NY', 'USA', '10012-2612', 15.84),
 (2, 30, '2017-01-03 00:00:00', '230 Elgin Street', 'Ottawa', 'ON', 'Canada', 'K2P 1L7', 9.9),
 (3, 40, '2017-01-05 00:00:00', '8, Rue Hanovre', 'Paris', 'None', 'France', '75002', 1.98),
 (4, 18, '2017-01-06 00:00:00', '627 Broadway', 'New York', 'NY', 'USA', '10012-2612', 7.92),
 (5, 27, '2017-01-07 00:00:00', '1033 N Park Ave', 'Tucson', 'AZ', 'USA', '85719', 16.83),
 (6, 31, '2017-01-10 00:00:00', '194A Chain Lake Drive', 'Halifax', 'NS', 'Canada', 'B3S 1C5', 1.98),
 (7, 49, '2017-01-12 00:00:00', 'Ordynacka 10', 'Warsaw', 'None', 'Poland', '00-358', 10.89),
 (8, 59, '2017-01-13 00:00:00', '3,Raj Bhavan Road', 'Bangalore', 'None', 'India', '560001', 9.9),
 (9, 18, '2017-01-18 00:00:00', '627 Broadway', 'New York', 'NY', 'USA', '10012-2612', 8.91),
 (10, 31, '2017-01-18 00:00:00', '194A Chain Lake Drive', 'Halifax', 'NS', 'Canada', 'B3S 1C5', 1.98)]

In [8]:
%%sql
SELECT *
  FROM track
 LIMIT 10

 * sqlite:///chinook.db
Done.


[(1, 'For Those About To Rock (We Salute You)', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 343719, 11170334, 0.99),
 (2, 'Balls to the Wall', 2, 2, 1, None, 342562, 5510424, 0.99),
 (3, 'Fast As a Shark', 3, 2, 1, 'F. Baltes, S. Kaufman, U. Dirkscneider & W. Hoffman', 230619, 3990994, 0.99),
 (4, 'Restless and Wild', 3, 2, 1, 'F. Baltes, R.A. Smith-Diesel, S. Kaufman, U. Dirkscneider & W. Hoffman', 252051, 4331779, 0.99),
 (5, 'Princess of the Dawn', 3, 2, 1, 'Deaffy & R.A. Smith-Diesel', 375418, 6290521, 0.99),
 (6, 'Put The Finger On You', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 205662, 6713451, 0.99),
 (7, "Let's Get It Up", 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 233926, 7636561, 0.99),
 (8, 'Inject The Venom', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 210834, 6852860, 0.99),
 (9, 'Snowballed', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 203102, 6599424, 0.99),
 (10, 'Evil Walks', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 263497, 8611245, 0.99)]

In [9]:
%%sql
SELECT *
  FROM genre

 * sqlite:///chinook.db
Done.


[(1, 'Rock'),
 (2, 'Jazz'),
 (3, 'Metal'),
 (4, 'Alternative & Punk'),
 (5, 'Rock And Roll'),
 (6, 'Blues'),
 (7, 'Latin'),
 (8, 'Reggae'),
 (9, 'Pop'),
 (10, 'Soundtrack'),
 (11, 'Bossa Nova'),
 (12, 'Easy Listening'),
 (13, 'Heavy Metal'),
 (14, 'R&B/Soul'),
 (15, 'Electronica/Dance'),
 (16, 'World'),
 (17, 'Hip Hop/Rap'),
 (18, 'Science Fiction'),
 (19, 'TV Shows'),
 (20, 'Sci Fi & Fantasy'),
 (21, 'Drama'),
 (22, 'Comedy'),
 (23, 'Alternative'),
 (24, 'Classical'),
 (25, 'Opera')]

In [10]:
%%sql
SELECT * 
  FROM invoice
 LIMIT 10

 * sqlite:///chinook.db
Done.


[(1, 18, '2017-01-03 00:00:00', '627 Broadway', 'New York', 'NY', 'USA', '10012-2612', 15.84),
 (2, 30, '2017-01-03 00:00:00', '230 Elgin Street', 'Ottawa', 'ON', 'Canada', 'K2P 1L7', 9.9),
 (3, 40, '2017-01-05 00:00:00', '8, Rue Hanovre', 'Paris', 'None', 'France', '75002', 1.98),
 (4, 18, '2017-01-06 00:00:00', '627 Broadway', 'New York', 'NY', 'USA', '10012-2612', 7.92),
 (5, 27, '2017-01-07 00:00:00', '1033 N Park Ave', 'Tucson', 'AZ', 'USA', '85719', 16.83),
 (6, 31, '2017-01-10 00:00:00', '194A Chain Lake Drive', 'Halifax', 'NS', 'Canada', 'B3S 1C5', 1.98),
 (7, 49, '2017-01-12 00:00:00', 'Ordynacka 10', 'Warsaw', 'None', 'Poland', '00-358', 10.89),
 (8, 59, '2017-01-13 00:00:00', '3,Raj Bhavan Road', 'Bangalore', 'None', 'India', '560001', 9.9),
 (9, 18, '2017-01-18 00:00:00', '627 Broadway', 'New York', 'NY', 'USA', '10012-2612', 8.91),
 (10, 31, '2017-01-18 00:00:00', '194A Chain Lake Drive', 'Halifax', 'NS', 'Canada', 'B3S 1C5', 1.98)]

In [11]:
%%sql
SELECT *
  FROM invoice_line
 LIMIT 10

 * sqlite:///chinook.db
Done.


[(1, 1, 1158, 0.99, 1),
 (2, 1, 1159, 0.99, 1),
 (3, 1, 1160, 0.99, 1),
 (4, 1, 1161, 0.99, 1),
 (5, 1, 1162, 0.99, 1),
 (6, 1, 1163, 0.99, 1),
 (7, 1, 1164, 0.99, 1),
 (8, 1, 1165, 0.99, 1),
 (9, 1, 1166, 0.99, 1),
 (10, 1, 1167, 0.99, 1)]

We can see that some tables have common columns. So they can be merged for the analysis.

## Finding the most popular genres

The Chinook record store has just signed a deal with a new record label, and we've been tasked with selecting the first three albums that will be added to the store, from a list of four:

* Hip-hop
* Punk
* Pop
* Blues

The record label specializes in artists from the USA, and they have given Chinook some money to advertise the new albums in the USA, so we're interested in finding out which genres sell the best in the USA.

In [12]:
%%sql
-- Select only tracks purchased in USA and number of purchases for the each track
WITH tracks_USA AS 
     (
     SELECT il.track_id,
            SUM(il.quantity) AS total
       FROM invoice i
      INNER JOIN invoice_line il ON il.invoice_id = i.invoice_id
      WHERE i.billing_country = 'USA'
      GROUP BY il.track_id
      ORDER BY total DESC
     )

SELECT g.name,
       SUM (total) tracks_sold
  FROM tracks_USA tu
 INNER JOIN track t ON t.track_id = tu.track_id
 INNER JOIN genre g ON g.genre_id = t.genre_id
 GROUP BY 1
 ORDER BY 2 DESC
 LIMIT 15

 * sqlite:///chinook.db
Done.


[('Rock', 561),
 ('Alternative & Punk', 130),
 ('Metal', 124),
 ('R&B/Soul', 53),
 ('Blues', 36),
 ('Alternative', 35),
 ('Pop', 22),
 ('Latin', 22),
 ('Hip Hop/Rap', 20),
 ('Jazz', 14),
 ('Easy Listening', 13),
 ('Reggae', 6),
 ('Electronica/Dance', 5),
 ('Classical', 4),
 ('Heavy Metal', 3)]

Base on the selling history in the USA, it is recommended for the store to purchase an album of artists from genres:
* Punk
* Blues
* Pop

## Performance of sales agents

Each customer for the Chinook store gets assigned to a sales support agent within the company when they first make a purchase. We have been asked to analyze the purchases of customers belonging to each employee to see if any sales support agent is performing either better or worse than the others.

In [13]:
%%sql
SELECT e.first_name || ' ' || e.last_name AS name,
       e.title,
       e.country,
       e.birthdate,
       ROUND(SUM(i.total), 2)
  FROM employee e
 INNER JOIN customer c ON c.support_rep_id = e.employee_id
 INNER JOIN invoice i ON i.customer_id = c.customer_id
 GROUP BY 1
 ORDER BY 5 DESC

 * sqlite:///chinook.db
Done.


[('Jane Peacock', 'Sales Support Agent', 'Canada', '1973-08-29 00:00:00', 1731.51),
 ('Margaret Park', 'Sales Support Agent', 'Canada', '1947-09-19 00:00:00', 1584.0),
 ('Steve Johnson', 'Sales Support Agent', 'Canada', '1965-03-03 00:00:00', 1393.92)]

All sales support agents are Canadian, the youngest one (Jane Peackock) has the best performance.

## Analyzing sales by country

Our next task is to analyze the sales data for customers from each different country. We have been given guidance to use the country value from the customers table, and ignore the country from the billing address in the invoice table.

In particular, we have been directed to calculate data, for each country, on the:

* total number of customers
* total value of sales
* average value of sales per customer
* average order value

In [14]:
%%sql
SELECT c.country,
       COUNT(distinct c.customer_id) AS customers_number,
       ROUND(SUM(i.total), 2) AS total_sales_value,
       ROUND(SUM(i.total)/COUNT(c.customer_id), 2) AS value_per_customer,
       ROUND(AVG(i.total), 2) AS average_order_value
  FROM customer c
 INNER JOIN invoice i ON i.customer_id = c.customer_id
 GROUP BY 1
 ORDER BY 2 DESC;

 * sqlite:///chinook.db
Done.


[('USA', 13, 1040.49, 7.94, 7.94),
 ('Canada', 8, 535.59, 7.05, 7.05),
 ('France', 5, 389.07, 7.78, 7.78),
 ('Brazil', 5, 427.68, 7.01, 7.01),
 ('Germany', 4, 334.62, 8.16, 8.16),
 ('United Kingdom', 3, 245.52, 8.77, 8.77),
 ('Portugal', 2, 185.13, 6.38, 6.38),
 ('India', 2, 183.15, 8.72, 8.72),
 ('Czech Republic', 2, 273.24, 9.11, 9.11),
 ('Sweden', 1, 75.24, 7.52, 7.52),
 ('Spain', 1, 98.01, 8.91, 8.91),
 ('Poland', 1, 76.23, 7.62, 7.62),
 ('Norway', 1, 72.27, 8.03, 8.03),
 ('Netherlands', 1, 65.34, 6.53, 6.53),
 ('Italy', 1, 50.49, 5.61, 5.61),
 ('Ireland', 1, 114.84, 8.83, 8.83),
 ('Hungary', 1, 78.21, 7.82, 7.82),
 ('Finland', 1, 79.2, 7.2, 7.2),
 ('Denmark', 1, 37.62, 3.76, 3.76),
 ('Chile', 1, 97.02, 7.46, 7.46),
 ('Belgium', 1, 60.39, 8.63, 8.63),
 ('Austria', 1, 69.3, 7.7, 7.7),
 ('Australia', 1, 81.18, 8.12, 8.12),
 ('Argentina', 1, 39.6, 7.92, 7.92)]

USA and Canada have the most costumers and total value coming from them. However, `average_order_value` is the highest in Czech Republic, Spain and Ireland - these 3 countries worth to be considered for business globalization.

## Albums vs Individual tracks

The Chinook store is setup in a way that allows customer to make purchases in one of the two ways:

* purchase a whole album
* purchase a collection of one or more individual tracks.

The store does not let customers purchase a whole album, and then add individual tracks to that same purchase (unless they do that by choosing each track manually). When customers purchase albums they are charged the same price as if they had purchased each of those tracks separately.

Management are currently considering changing their purchasing strategy to save money. The strategy they are considering is to purchase only the most popular tracks from each album from record companies, instead of purchasing every track from an album.

We have been asked to find out what percentage of purchases are individual tracks vs whole albums, so that management can use this data to understand the effect this decision might have on overall revenue.

In [15]:
%%sql

-- identify album for each invoice

WITH invoice_album AS
       (
      SELECT il.invoice_id,                                     
               t.album_id
        FROM invoice_line il
        INNER JOIN track t ON t.track_id = il.track_id
        GROUP BY 1),

-- identify if the whole album was purchased or just tracks

    album_vs_tracks AS (
        SELECT ia.*,
               CASE
               WHEN (
             SELECT il.track_id 
               FROM invoice_line il
              WHERE il.invoice_id = ia.invoice_id
              EXCEPT                                          
              SELECT t.track_id 
              FROM track t
              WHERE t.album_id = ia.album_id) IS NULL
                             
              AND (
            SELECT t.track_id 
              FROM track t
             WHERE t.album_id = ia.album_id                                         
             EXCEPT                                          
             SELECT il.track_id 
             FROM invoice_line il
             WHERE il.invoice_id = ia.invoice_id) IS NULL
                                   
             THEN "Albums"
             ELSE "Tracks"
             END AS purchase                             
             FROM invoice_album ia)
    
SELECT purchase,
       COUNT(*) num_invoices,
       ROUND(COUNT(*) * 100.0 /(SELECT COUNT(*)
                                  FROM invoice), 0) pct
  FROM album_vs_tracks
 GROUP BY 1
 ORDER BY 2 DESC

 * sqlite:///chinook.db
Done.


[('Tracks', 500, 81.0), ('Albums', 114, 19.0)]

Since 81% of purchases are tracks, we recommend the shop to follow the proposed strategy of purchasing only the most popular tracks from each album from record companies instead of purchasing every track from the album.

## Conclusions

In this project we worked with imaginary music store's database. Using SQL the required information has been extracted to answer the following questions:

* What genres are the most popular.
* Who is the best performer along the sales agents.
* Which countries have the biggest number of customers and value coming from them.
* Does it worth to purchase only the most popular tracks from each album from record companies, instead of purchasing every track from an album.